core/app/models/spree/calculator/promotion/distributed_amount.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require_dependency 'spree/calculator'
+
+module Spree
+  # This is a calculator for line item adjustment actions. It accepts a line item
+  # and calculates its weighted adjustment amount based on the value of the
+  # preferred amount and the price of the other line items. More expensive line
+  # items will receive a greater share of the preferred amount.
+  class Calculator::Promotion::DistributedAmount < Calculator
```

Comment:
What's the preferred way to write the namespaces? Here we have

```
module Spree
  class Calculator::Promotion::DistributedAmount < Calculator
```

and on something like a shipping calculator https://github.com/solidusio/solidus/blob/master/core/app/models/spree/calculator/shipping/flat_rate.rb we have

```
module Spree
  module Calculator::Shipping
    class FlatRate < ShippingCalculator
```

with the modules on their own line and the class alone at the bottom.

Just wondering what our preferred style should be on these.